### PR TITLE
[release-1.22] security adv, release notes 1.22.1 and 1.21.4

### PR DESCRIPTION
**Pull request by dhawton**
## Description

Security advisory 2024-0005 and release notes for 1.22.1 and 1.21.4. Also fixed the missed removals for 1.20 that is EOL as of the 25th. Will manually cherrypick to `master` after releases finalized.

## Reviewers

- [ ] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

**Comment by dhawton** [comment]
doc.test.profile-ambient_istio.io_release-1.22 -- this is a new test that is not required, and not setup for release-1.22.

**Comment by kfaseela** [comment]
> doc.test.profile-ambient_istio.io_release-1.22 -- this is a new test that is not required, and not setup for release-1.22.

this has been added as per request from @craigbox but it is not optional as of now. It is needed for his pending cherrypick PR

**Comment by dhawton** [comment]
> > doc.test.profile-ambient_istio.io_release-1.22 -- this is a new test that is not required, and not setup for release-1.22.
> 
> this has been added as per request from @craigbox but it is not optional as of now. It is needed for his pending cherrypick PR

It is marked as optional

**Comment by kfaseela** [comment]
> > > doc.test.profile-ambient_istio.io_release-1.22 -- this is a new test that is not required, and not setup for release-1.22.
> > 
> > 
> > this has been added as per request from @craigbox but it is not optional as of now. It is needed for his pending cherrypick PR
> 
> It is marked as optional

sorry I meant it "is optional" ;)

File: content/en/news/releases/1.21.x/announcing-1.21.4/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+linktitle: 1.21.4
+subtitle: Patch Release
+description: Istio 1.21.4 patch release.
+publishdate: 2024-06-27
+release: 1.21.4
+---
+
+This release implements the security updates described in our 27th of June post, [`ISTIO-SECURITY-2024-005`](/news/security/istio-security-2024-005) along with bug fixes to improve robustness.
+
+This release note describes what’s different between Istio 1.21.3 and 1.21.4.
```
Comment (kfaseela) [inline content/en/news/releases/1.21.x/announcing-1.21.4/index.md:12]:
```suggestion
This release note describes what is different between Istio 1.21.3 and 1.21.4.
```

File: content/en/news/releases/1.22.x/announcing-1.22.2/index.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+linktitle: 1.22.2
+subtitle: Patch Release
+description: Istio 1.22.2 patch release.
+publishdate: 2024-06-27
+release: 1.22.2
+---
+
+This release implements the security updates described in our 27th of June post, [`ISTIO-SECURITY-2024-005`](/news/security/istio-security-2024-005) along with bug fixes to improve robustness.
+
+This release note describes what’s different between Istio 1.22.1 and 1.22.2.
```
Comment (kfaseela) [inline content/en/news/releases/1.22.x/announcing-1.22.2/index.md:12]:
```suggestion
This release note describes what is different between Istio 1.22.1 and 1.22.2.
```

File: content/en/news/releases/1.22.x/announcing-1.22.2/index.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+- **Added** `gateways.securityContext` to manifests to provide an option to customize the gateway `securityContext`.
+  ([Issue #49549](https://github.com/istio/istio/issues/49549))
+
+- **Added** a new option in ztunnel to completely disable IPv6, to enable running on kernels with IPv6 disabled.
+
+- **Fixed** an issue where `istioctl analyze` returned IST0162 false positives.
+  ([Issue #51257](https://github.com/istio/istio/issues/51257))
+
+- **Fixed** `ENABLE_ENHANCED_RESOURCE_SCOPING` not being part of helm compatibility profiles for Istio 1.20/1.21.
```
Comment (kfaseela) [inline content/en/news/releases/1.22.x/announcing-1.22.2/index.md:28]: 1.20 makes sense here?

File: content/en/news/releases/1.22.x/announcing-1.22.2/index.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+- **Fixed** false positives in IST0128 and IST0129 when `credentialName` and `workloadSelector` were set.
+  ([Issue #51567](https://github.com/istio/istio/issues/51567))
+
+- **Fixed** an issue where JWKS fetched from URIs were not updated promptly when there are errors fetching other URIs.
+  ([Issue #51636](https://github.com/istio/istio/issues/51636))
+
+- **Fixed** an issue causing `workloadSelector` policies to apply to the wrong namespace in ztunnel.
+  ([Issue #51556](https://github.com/istio/istio/issues/51556))
+
+- **Fixed** a bug causing `discoverySelectors` to accidentally filter out all `GatewayClasses`.
```
Comment (kfaseela) [inline content/en/news/releases/1.22.x/announcing-1.22.2/index.md:42]: this one onwards, there are no linked GitHub issues

Comment (dhawton) [reply content/en/news/releases/1.22.x/announcing-1.22.2/index.md:42]: Not in release notes for them

File: content/en/news/releases/1.22.x/announcing-1.22.2/index.md (hunk 3 above)
Comment (dhawton) [reply content/en/news/releases/1.22.x/announcing-1.22.2/index.md:28]: I left to leave trails for people who upgrade from 1.20 to here

**Comment by kfaseela** [comment]
LGTM. May be @craigbox can have a final look and merge.

**Review by howardjohn — COMMENTED** [review]
LGTM

**Review by craigbox — APPROVED** [review]

**Pull request merged**